test/NuGet.Core.Tests/NuGet.Packaging.Test/NoRefOrLibFolderInPackageRuleTests.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Commands;
+using NuGet.Test.Utility;
+using Xunit;
+using NuGet.Packaging.Rules;
+using Moq;
+using System.Xml.Linq;
+using System.Collections;
+using System.Runtime.CompilerServices;
+using System.Reflection;
+
+namespace NuGet.Packaging.Test
+{
+    public class NoRefOrLibFolderInPackageRuleTests
+    {
+        [Theory]
+        [MemberData("WarningRaisedData", MemberType = typeof(FileSource))]
+        public void WarningRaisedWhenLibOrRefFolderWithTFMDataIsNotFoundAndBuildFolderIsFound(string[] files)
+        {
+            files = files.ToArray();
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.True(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+
+        [Theory]
+        [MemberData("WarningNotRaisedData", MemberType = typeof(FileSource))]
+        public void WarningNotRaisedWhenLibOrRefFolderWithTFMDataIsFound(string[] files)
+        {
+            files = files.ToArray();
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+    }
+
+    public static class FileSource
+    {
+        public static readonly List<object[]> WarningNotRaisedData
+                = new List<object[]>
+                {
+                    new object[] { new string[] {"ref/net45/test.dll", "test.dll"} },
+                    new object[] { new string[] {"lib/net45/test.dll", "test.dll"} },
+                    new object[] { new string[] {"build/net45/test.props", "ref/net45/test.dll"} },
+                    new object[] { new string[] {"build/net45/test.props", "lib/net45/test.dll"} },
+                    new object[] { new string[] {"ref/net45/test.dll"} },
+                    new object[] { new string[] {"lib/net45/test.dll"} },
+                    new object[] { new string[] {"build/any/test.props"} },
+                    new object[] { new string[] {"build/native/test.props"} }
+                };
+
+        public static readonly List<object[]> WarningRaisedData
+            = new List<object[]>
+            {
+                new object[] { new string[] {"build/net45/test.props"} },
+                new object[] { new string[] {"build/net45/test.props", "build/any/test.props", "build/native/test.props"} },
+                new object[] { new string[] {"build/net45/test.props", "content/net45/test.props"} },
+                new object[] { new string[] {"build/net45/test.props", "test.props"} }
+            };
+    }
+
+    internal class TestPackageFile : IDisposable
```

Comment:
This class doesn't appear to be used any longer, so should be deleted (unless I missed a reference to it somewhere)

Reply:
Well do you think we could use it for anything?

Reply:
You should add at least 1 test, making sure that the 
```cs
Validate(PackageArchiveReader reader)
```
code path works.

Reply:
@nkolev92 it's out of scope for this PR, but what do you think about extracting an interface for `PackageArchiveReader`, so it can be mocked for tests that use it and don't have create files on disk?